source/blog/042-wide-gamut-colors-in-sass.md

```diff
@@ -0,0 +1,279 @@
+
+## Working with gamut boundaries
+
+So what happens when you go outside the gamut of a given display? Browsers are still debating the details, but everyone agrees we have to display *something*:
+
+- Currently, browsers convert every color into `red`, `green`, and `blue` channels for display. If any of those channels are too high or two low for a given screen, they get *clamped* at the highest or lowest value allowed. This is often referred to as 'channel clipping'. It keeps the math simple, but it can have a weird effect on both the `hue` and `lightness` if some channels are clipped more than others.
+- The CSS specification says that preserving `lightness` should be the highest priority, and provides an algorithm for reducing `chroma` until the color is in gamut. That's great for maintaining readable text, but it's more work for browsers, and it can be surprising when colors suddenly lose their vibrance.
+- There's been some progress on a compromise approach, reducing `chroma` to get colors inside the `rec2020` gamut, and clipping from there.
+
+Since browser behavior is still unreliable, and some color spaces (*cough* `oklch`) can easily launch us out of any available gamut, it can be helpful to do some gamut management in Sass.
+
+We can use `color.is-in-gamut()` to test if a particular color is in a given gamut. Like our other color functions, this will default to the space the color is defined in, but we can provide a `$space` parameter to test it against a different gamut:
+
+```scss
+@use 'sass:color';
+$extra-pink: color(display-p3 0.951 0.457 0.7569);
+
+// result: true, for display-p3 gamut
+$in-p3: color.is-in-gamut($extra-pink);
+
+// result: false, for srgb gamut
+$in-srgb: color.is-in-gamut($extra-pink, srgb);
+```
+
+We can also use the `color.to-gamut()` function to explicitly move a color so that it is in a particular gamut. Since there are several options on the table, and no clear sense what default CSS will use long-term, this function currently requires an explicit `$method` parameter. The current options are `clip` (as is currently applied by browsers) or `local-minde` (as is currently specified):
+
+```scss
+@use 'sass:color';
+$extra-pink: color(display-p3 0.951 0.457 0.7569);
+
+// result: ???
+$clip-to-srgb: color.to-gamut($extra-pink, srgb, clip);
+
+// result: ???
+$clip-to-srgb: color.to-gamut($extra-pink, srgb, local-minde);
+```
+
+All legacy and RGB-style spaces represent bounded gamuts of color. Since mapping colors into gamut is a lossy process, it should generally be left to browsers or done with caution. For that reason, out-of-gamut channel values are maintained by Sass whenever possible, even when converting into gamut-bounded color spaces. The only exception is that `hsl` and `hwb` color spaces are not able to express out-of-gamut color, so converting colors into those spaces will gamut-map the colors as well.
```

Comment:
The last sentence here is no longer true. We now preserve out-of-gamut colors _everywhere_.